### ISSUE TITLE
JS templating in Backend/Frontend ?

### ISSUE THREAD

**Issue by arnisjuraga** [copona/copona#32]
Need your opinion on Javascript source code replacement and refactoring for catalog and admin templates. 

Right now - a lot of code is included in .tpl files directly. 
They uses PHP variables, and as we do not support any template engine yet for HTML (Opencart says, they will soon introduce Twig)  - php variables into .js code is a headache.

What would be the ideas for javascript exclusion from .TPL files?

I have seen excellent examples done in Vanilla .js, which are EXCELLENT! ... but takes time to read and understand, so would not be the best choice for Copona scope, because of simplicity in code.

Keep in mind - the solution must be as simple as possible to be available for as many developers we can.

**Comment by prhost** [comment]
In fact I think it is more lack of organization on the part of Opencart. I believe that the php used in js could be done by other ways, as instead of filling in the id of a product by php in js, it can catch it by date attribute once only printed on some part of the html page. So I guess it would be more of a refactoring like you said. There is a technique that you can add js in any part of the template the function sends the js to its correct location as the end of the footer.

**Comment by arnisjuraga** [comment]
Wouldn't it be better to move js completely out of templates?

**Comment by prhost** [comment]
Unfortunately this would limit developers a lot, at least we make a way to inject js in a more organized and modular way, and I think we have to give the possibility of working with any js framework, forcing the developer to learn angularjs, vuejs or vanillajs It'll be a good idea. Now give all possible tools for the developer to create your theme any way you want, this is cool (:

Another thing, Angular, vue, Vanilla, use API, would have to convert the entire platform to API, which is being discussed in issue #26

**Comment by arnisjuraga** [comment]
No, I mean - just an "approach", so - to move all .js code from templates to one place, or to multiple .js files, wchich are contcatenated. 
The reason for using .js in templates is only for PHP variables. If we crate a way to exchange them between php and .js easy, then - let's move it outside.

**Comment by prhost** [comment]
A simple way without using tools is to refactor the whole view that has php code mixed with javascript and put the php in html or by ajax.

Example:

```html
<div data-product-id="<?php echo $product_id; ?>">
....
</div>
...
<script>
var product_id = $('div').data('product-id');
</script>
```
